CineElorrietaMVC/src/main/java/Vista/PanelFacturas.java

```diff
@@ -211,38 +237,120 @@ public void actionPerformed(ActionEvent arg0) {
 				String cantidad = textCantidad.getText();
 				try {
 					producto = (String) listaProductos.getSelectedValue();
-					productoAnadir = controladorPanelFacturas.accionadoBotonAnnadirProducto(producto);
-					existeProd = true;
-				}
-				catch(Exception e) {
+					if(producto != null) {
+						existeProd = true;
+					}
+					
+				} catch (Exception e) {
 					System.out.println("No se ha seleccionado un producto");
 					lblError.setText("No se ha escogido un producto");
 				}
 				if (existeProd) {
 					try {
-						annadidos.addElement(controladorPanelFacturas.cantidadProducto(cantidad, productoAnadir));
-						textTotal.setText(controladorPanelFacturas.cantidadTotal(cantidad,  textTotal.getText(), producto));
-					}
-					catch(Exception e) {
+						if (controladorPanelFacturas.existeProducto(producto) == -1) {
+							productoAnadir = controladorPanelFacturas.accionadoBotonAnnadirProducto(producto);
+							annadidos.addElement(controladorPanelFacturas.cantidadProducto(cantidad, productoAnadir));
+							textTotal.setText(
+									controladorPanelFacturas.cantidadTotal(cantidad, textTotal.getText(), producto));
+							lblError.setText("");
+						} else {
+							String yaAnnadido = annadidos.get(controladorPanelFacturas.existeProducto(producto));
+							annadidos.set(controladorPanelFacturas.existeProducto(producto), controladorPanelFacturas
+									.cambiarCantidadProductos(yaAnnadido, Integer.parseInt(cantidad)));
+							String total = Double
+									.toString(Double.parseDouble(textTotal.getText()) + (Double.parseDouble(cantidad)
+											* controladorPanelFacturas.cogerPrecioString(producto)));
+							textTotal.setText(total);
+						}
+					} catch (Exception e) {
 						System.out.println("El campo cantidad no contiene un entero");
 						lblError.setText("No se ha introducido una cantidad");
 					}
 				}
 			}
 		};
 	}
-	
+
+	private ActionListener listenerBotonFinalizar(ControladorPanelFacturas controladorPanelFacturas) {
+		return new ActionListener() {
+			public void actionPerformed(ActionEvent arg0) {
+				System.out.println("Ejecutando evento Boton Finalizar");
+				// Comprobamos si los campos DNI, Nombre, Apellido y si hay algun articulo metido  
+				if (Double.parseDouble(textTotal.getText()) > 0 && 
+						controladorPanelFacturas.comprobarNif(textNIF.getText()) &&
+						controladorPanelFacturas.comprobarFormatoApellido(textNombre.getText()) &&
+						controladorPanelFacturas.comprobarFormatoApellido(textApellido.getText())
+						) {
+					
+					
+					// insertar datos en actividad
+					controladorPanelFacturas.insertarActividad(Integer.parseInt(textFieldNumTrans.getText()),
+							controladorPanelFacturas.devolverFechaFormateada(textFieldFecha.getText()),
```

Comment:
Esta llamada al controlador sobra, podéis pasar la fecha como string y formatearla allí mismo

Reply:
Corregido.